src/tools/plan-generator.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {Recipe} from '../runtime/recipe/recipe.js';
+import {Type} from '../runtime/type.js';
+import {Particle} from '../runtime/recipe/particle.js';
+import {KotlinGenerationUtils} from './kotlin-generation-utils.js';
+import {HandleConnection} from '../runtime/recipe/handle-connection.js';
+import {StorageKey} from '../runtime/storageNG/storage-key.js';
+
+const ktUtils = new KotlinGenerationUtils();
+
+export class PlanGeneratorError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = 'PlanGeneratorError';
+  }
+}
+
+/** Generates plan objects from resolved recipes. */
+export class PlanGenerator {
+  constructor(private resolutions: Recipe[], private scope: string = 'arcs.core.data') {
+  }
+
+  /** Generates a Kotlin file with plan classes derived from resolved recipes. */
+  generate(): string {
+    const planOutline = [
+      this.fileHeader(),
+      this.createPlans().join('\n'),
+      this.fileFooter()
+    ];
+
+    return planOutline.join('\n');
+  }
+
+  /** Converts a resolved recipe into a `Plan` object. */
+  createPlans(): string[] {
+    return this.resolutions.map(recipe => {
+      const planName = `${recipe.name.replace(/[rR]ecipe/, '')}Plan`;
```

Comment:
Why? Why this rename? You are removing "recipe" word from user define name of a recipe. If you do this, you need to document it etc. it's unnecessary. Users should not put "recipe" as the name of their recipes FWIW, they should just say what it does, e.g.

```
recipe IngestOverviewEvents
  (...)
```

Reply:
With this rename if developer for example does:
```
recipe CookingRecipesHandler
  (...)
```
It will be translated in Kotlin to `CookingsHandlerPlan`, you see my point.

Reply:
Fair, this is an overreach.